### ISSUE TITLE
Validation has violation when defining named example on optional query parameter in RAML overlay

### ISSUE THREAD

**Issue by wes-goulet** [aml-org/amf#361]
1. create an endpoint with an optional query parameter

```yaml
/test-entity:
  get:
    responses:
      '200':
        body:
          application/json:
            type: TestEntity
    queryParameters:
      q1:
        required: false
        type: string
```

2. use an overlay.raml to add a named example to an optional query parameter

```yaml
/test-entity:
  get:
    queryParameters:
      q1:
        examples:
          myexample1: 'foo'
```

2. use amf-client-js to validate with RAML validation profile

```js
const { results } = await amf.AMF.validate(
    model,
    amf.ProfileNames.RAML,
    amf.MessageStyles.RAML
);
```

Expected:
No violations (`results is empty`)

Actual:
Violation occurs with ID: `http://a.ml/vocabularies/amf/core#resolution-validation` and message: `Property 'http://a.ml/vocabularies/apiContract#required' in 'Parameter' is not allowed`

The bug only happens on optional query parameters. Adding a named example to a required query parameter in an overlay does not cause a violation.

I have created a simple repro repo at https://github.com/wes566/amf-named-example-in-optional-query-param-in-overlay-bug-repo

**Comment by nschejtman** [comment]
> Hi @tomsfernandez - if I set the `required` property in the overlay then I agree that this would be an obvious violation of the spec. But I'm not setting the `required` property in the overlay, I'm intentionally leaving it unset because I don't want to modify any aspect of the base RAML.
> 
> I guess this is a feature request: it would be great if AMF parser didn't attempt to apply defaults to unset properties in an overlay. Just let me "overlay" my extra stuff and don't make me re-define properties (like `required`) that my overlay doesn't care about.
> 
> Thanks!

@wes566 created APIMF-2264 to track internally this enhancement request in our Jira

**Comment by AgustinBettati** [comment]
Hello @wes566, this issue has been fixed (https://github.com/aml-org/amf/pull/856) and will be available in the coming AMF release (4.7.1) due for next week.
Thanks!
